### DISCUSSION TITLE
./cmi mail should open a GUI with mails & clicking mail opens up in a book.

### DISCUSSION THREAD

**Discussion by TomLewis** [Zrips/CMI#4086] [Feature Requests]
When reading ./cmi mail, all mail should open in a GUI to not be lost in chat when chat is moving really fast, when you click one of the mails in the GUI it opens up the message in a book.

This would make reading ./cmi mail MUCH more user friendly and it will not get pushed down in-chat as chat makes it hard to read.

**Comment by CloudeLecaw** [comment]
ah you mean a bookGUi where the email of the sender so you dont need to read it in chat? Nice Idea
The book just should be a bookGUI - so when closing then the book should not be given to inventory

i like your idea :)

**Comment by TomLewis** [comment]
Yeah, he already has `/openbook` so it shouldn't be too hard to implement.

**Comment by CloudeLecaw** [comment]
your not right.
open books opens an allready existsing book
getbook opens ctesxts

so there would need to be a new "converter" that converts all mails to bookformat
and there needs also to be an option for all that dont want this
well this is not so easy to create how it may look ;)

**Comment by TomLewis** [comment]
> 
> 
> your not right.
> open books opens an allready existsing book
> getbook opens ctesxts
> 
> so there would need to be a new "converter" that converts all mails to bookformat
> and there needs also to be an option for all that dont want this
> well this is not so easy to create how it may look ;)

Thats what I meant, there is a command to open a ctext already, the **CODE TO OPEN BOOKS** exists already so it wouldn't be hard to implement. You know what I meant, you are just being an asshole again.
You are being pedantic for no reason on all my tickets, please do not reply to any more of my tickets, you are not helpful.

**Comment by CloudeLecaw** [comment]
i see u never created a book with ctext ;) else you would know that u need to create a special ctext extact designt to the book (because a book just have a few words for each line)
and its totally different from the databse entry of a mail

but ok^^

the **asshole** now tells you that u may have to rethink about this a bit
adding text from a database somehwere else is not so hard
but building it in a book means in this case it has to be a complete new module and this is a bit harder to write because u have to design all aspects of the book GUI too. saving an mail as ctext is no solution here. also taking this 1:1 to the database entries of mails is not possible that easy.
but hey.. if u say its not so hard? why u dont code a simple plugin like that by yourself? ;D
_The asshole is moving now to an other thread.. have a nice day_

**Comment by TomLewis** [comment]
> 
> 
> i see u never created a book with ctext ;) else you would know that u need to create a special ctext extact designt to the book (because a book just have a few words for each line)
> and its totally different from the databse entry of a mail
> 
> but ok^^
> 
> the **asshole** now tells you that u may have to rethink about this a bit
> adding text from a database somehwere else is not so hard
> but building it in a book means in this case it has to be a complete new module and this is a bit harder to write because u have to design all aspects of the book GUI too. saving an mail as ctext is no solution here. also taking this 1:1 to the database entries of mails is not possible that easy.
> but hey.. if u say its not so hard? why u dont code a simple plugin like that by yourself? ;D
> _The asshole is moving now to an other thread.. have a nice day_

What the hell is your problem? 

I use ctext for books using `/openbook` for years.

I have told you to not reply on my tickets as you are just an asshole with bad ideas, stay off my tickets.

**Comment by CloudeLecaw** [comment]
hm this is like a recuting to answer ;)

and good that you know what ctext is ;)
well this still does not made it possible to easy to add this for mails like u wished ;)

**Comment by TomLewis** [comment]
> 
> 
> hm this is like a recuting to answer ;)
> 
> and good that you know what ctext is ;)
> well this still does not made it possible to easy to add this for mails like u wished ;)

Stop replying to my tickets you have no idea what you are talking about, you are just replying to tickets like you have any shred of an idea how things work.
I cant make this any simpler for you, **THE CODE EXISTS TO OPEN BOOKS FROM A FILE ALREADY**. What this means, is that it would not be difficult to adapt this for other saved data. Its not that hard to understand.

**Comment by 456dev** [comment]
The code for opening books from a plain text file will not be the same for opening a book from a salute/SQL database.

Their is a built-in (bukkit) method for opening a book to the player. You *just* need a book to open. 
The mail storage is not the same as the ctext storage, and would therefore need to be converted differently, with (afaik) very little to no code reuse

You *could* build a book, say with line 1 the sender, and a gap then the rest of the message
But it's still much harder than reusing "**THE CODE** (that) **EXISTS TO OPEN BOOKS FROM A FILE ALREADY**"

**Comment by TomLewis** [comment]
> 
> 
> The code for opening books from a plain text file will not be the same for opening a book from a salute/SQL database.
> 
> Their is a built-in (bukkit) method for opening a book to the player. You _just_ need a book to open.
> The mail storage is not the same as the ctext storage, and would therefore need to be converted differently, with (afaik) very little to no code reuse
> 
> You _could_ build a book, say with line 1 the sender, and a gap then the rest of the message
> But it's still much harder than reusing "**THE CODE** (that) **EXISTS TO OPEN BOOKS FROM A FILE ALREADY**"

I would assume mail is loaded into memory, most likley an array when a player logs in.
An SQL query most likley pulls the player row when they log in currently to pull their data. 
CMI saves the mails of a player in this single row with all their other information, so its most likely ALREADY IN MEMORY, already loaded as an array, so instead of having to pull a files contents, its less work, you are pulling data from an array thats already loaded for that player and instead of displaying it in chat, you display it in a book, for which I have already told you, the code exists to open a fake book and display any text in it.

Do not try and come across as technical when you obviously have no idea how things work.

**Comment by 456dev** [comment]
It shouldn't be the pulling of the day, bit the fact that they are in two different formats.
Can you have authors / sender's of ctexts? No
Do you need a built in button to reply? No

"Display any text" is the problem. The mail isnt in "any text" format, its a structure with some static and dynamic pieces
